schema/db2/cadence/schema.sql

```diff
@@ -0,0 +1,257 @@
+);
+
+CREATE TABLE cadence.task_lists (
+  shard_id INT NOT NULL,
+  domain_id CHAR(16) NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  task_type SMALLINT NOT NULL, -- {Activity, Decision}
+  --
+  range_id BIGINT NOT NULL,
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16) NOT NULL,
+  PRIMARY KEY (shard_id, domain_id, name, task_type)
+);
+
+CREATE TABLE cadence.replication_tasks (
+  shard_id INT NOT NULL,
+  task_id BIGINT NOT NULL,
+  --
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16) NOT NULL,
+  PRIMARY KEY (shard_id, task_id)
+);
+
+CREATE TABLE cadence.replication_tasks_dlq (
+  source_cluster_name VARCHAR(255) NOT NULL,
+  shard_id INT NOT NULL,
+  task_id BIGINT NOT NULL,
+  --
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16) NOT NULL,
+  PRIMARY KEY (source_cluster_name, shard_id, task_id)
+);
+
+CREATE TABLE cadence.timer_tasks (
+  shard_id INT NOT NULL,
+  visibility_timestamp TIMESTAMP NOT NULL,
+  task_id BIGINT NOT NULL,
+  --
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16) NOT NULL,
+  PRIMARY KEY (shard_id, visibility_timestamp, task_id)
+);
+
+CREATE TABLE cadence.activity_info_maps (
+-- each row corresponds to one key of one map<string, ActivityInfo>
+  shard_id INT NOT NULL,
+  domain_id CHAR(16) NOT NULL,
+  workflow_id VARCHAR(255) NOT NULL,
+  run_id CHAR(16) NOT NULL,
+  schedule_id BIGINT NOT NULL,
+--
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16),
+  last_heartbeat_details BLOB,
+  last_heartbeat_updated_time TIMESTAMP NOT NULL,
+  PRIMARY KEY (shard_id, domain_id, workflow_id, run_id, schedule_id)
+);
+
+CREATE TABLE cadence.timer_info_maps (
+  shard_id INT NOT NULL,
+  domain_id CHAR(16) NOT NULL,
+  workflow_id VARCHAR(255) NOT NULL,
+  run_id CHAR(16) NOT NULL,
+  timer_id VARCHAR(255) NOT NULL,
+--
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16),
+  PRIMARY KEY (shard_id, domain_id, workflow_id, run_id, timer_id)
+);
+
+CREATE TABLE cadence.child_execution_info_maps (
+  shard_id INT NOT NULL,
+  domain_id CHAR(16) NOT NULL,
+  workflow_id VARCHAR(255) NOT NULL,
+  run_id CHAR(16) NOT NULL,
+  initiated_id BIGINT NOT NULL,
+--
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16),
+  PRIMARY KEY (shard_id, domain_id, workflow_id, run_id, initiated_id)
+);
+
+CREATE TABLE cadence.request_cancel_info_maps (
+  shard_id INT NOT NULL,
+  domain_id CHAR(16) NOT NULL,
+  workflow_id VARCHAR(255) NOT NULL,
+  run_id CHAR(16) NOT NULL,
+  initiated_id BIGINT NOT NULL,
+--
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16),
+  PRIMARY KEY (shard_id, domain_id, workflow_id, run_id, initiated_id)
+);
+
+CREATE TABLE cadence.signal_info_maps (
+  shard_id INT NOT NULL,
+  domain_id CHAR(16) NOT NULL,
+  workflow_id VARCHAR(255) NOT NULL,
+  run_id CHAR(16) NOT NULL,
+  initiated_id BIGINT NOT NULL,
+--
+  data BLOB NOT NULL,
+  data_encoding VARCHAR(16),
+  PRIMARY KEY (shard_id, domain_id, workflow_id, run_id, initiated_id)
+);
+
+CREATE TABLE cadence.buffered_replication_task_maps (
```

Comment:
This table is not being used.

Reply:
it's in sync with MySQL and PostgreSQL schemas